documentation/source/TabCompletion.md

```diff
@@ -0,0 +1,42 @@
+# Tab Completion
+
+SuperGradients configuration system uses Hydra, so it is possible to enable [tab completion](https://hydra.cc/docs/tutorials/basic/running_your_app/tab_completion/) feature.
+
+## Installation
+
+To enable tab completion you need to run following commands:
+
+Example for train_from_recipe.py (Command to be executed from `src/super_gradients`)
+
+`eval "$(python train_from_recipe.py -sc install=bash)"`
+
+If you're using ZSH (Mac OS X), then ensure that you have the following lines in your `.zshrc`:
+```
+autoload -Uz compinit
+compinit
+autoload -Uz bashcompinit && bashcompinit
+```
+
+See [Tab Completion](https://hydra.cc/docs/tutorials/basic/running_your_app/tab_completion/) for more information.
+
+## Caveats & Limitations
+
+* Tab completion is not supported on Windows.
+* You need to enable tab completion explicitly for each entry point in your application. In other words, if your project.
+* Tab completion is not supported for running train_from_recipe using `-m` form: `python -m super_gradients.train_from_recipe ...`.
```

Comment:
Why not actually?
Cant we just do the same wherever SG package is stored in ? Or is it a problem with the fact that its being ran as a module?